.travis.yml

```diff
@@ -57,24 +33,45 @@ env:
     # DIGITALOCEAN_ACCESS_TOKEN (from the service user's account and https://cloud.digitalocean.com/account/api/tokens)
     - secure: 'Ttp0WJa8C26wG1Lo5RGrab9kENZE/eDuzgifKogsywutRX/klzPDBoSvZpa7ZljVgEgoMGb7kNrTMx8TdvWphw7sy03ZiShhIS2I+w8jO+R1aSlFxAybwCpyg40I1bdEq5ctEh5tDC7YbkwD026ysJjVfZ6tU6yNiytbbYIPR0aXfHVlbF4IThg6yurP2JGsYOA37p9qyHNtcDWa/qkXFauiwJhA9KneKbSFAGKbBE5v4B1Us5hbciCHIQlsMOHaRgK6NBOJCiqL5WJyWM3Idsh8WMQxxCOV5FS5lLi9GBceF79NCOmE8NdG240/xd1HjSxuFkkJDIe/YQQoehrr9ql8JhvxZ44Mef//wTscoo6ik6/WQ/wxdd8JVJ+kvPEiyen6QqaFNQgUc1Uo8J/TQIk2qsCvxILXcE+a4vwlc09Au81OiuSqkw8QVjmMCjw6b6j/Y/bY/KXc9qDr7vU+yn+WhFMSN8NVFFgkjwAL8xQgcHgpue9wIg93k+9n1ufKxZtClDcnExBRKjjknyG+3L+PnznGBJy2bh+dgaSFnSdgO0HwzilWww2jdJ/gir/Iv6Pm+moum2MgW6B7ZRIXEISyY7ySrzCZcN0MlVjCA6u8SpePd68JPsKWtsQTbT/2WPo8z2GAhz6y2dYJGSoUzNd78KtgTdOxQ0r7LfZCie8='
 
-before_script:
-  - bash setup_postgresql_local.sh
-
 jobs:
   include:
     # Run security audit, linting, unit tests, and integration tests
     - stage: 'server test'
+      language: node_js
+      # node_js: node version specified in .nvmrc
+      install: 
+        - cd server
+        - npm ci
+      # install Redis
+      services:
+        - redis-server
+      addons:
+        # install PostgreSQL
+        postgresql: '12'
+        apt:
+          packages:
+            - postgresql-12
+            - postgresql-client-12
+      before_script:
+        - bash setup_postgresql_local.sh
```

Comment:
Moving this here saved a whole bunch of time and stopped a problem where the `firmware` stage would fail to run `npm ci`. I think this was because its language is `cpp` which doesn't install all the same things as when the language is `node_js`